### PR TITLE
DOC: Update indexing docs. 

### PR DESCRIPTION
**Pull request by charris**
Cleanup of #4986

Update docs for boolean array indexing and nonzero order.
Add links to row-major and column-major terms where they appear.

Cleanup squashes commits and rebases on master. Not yet ready to merge, but easier to review and modify here against current master.

### CODE REVIEW

File: numpy/doc/indexing.py [hunk 1]
```diff
@@ -229,10 +228,13 @@
...
  array([21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34])
 
-The result is a 1-D array containing all the elements in the indexed
-array corresponding to all the true elements in the boolean array. As
-with index arrays, what is returned is a copy of the data, not a view
-as one gets with slices.
+Unlike in the case of integer index arrays, in the boolean case, the
+result is a 1-D array containing all the elements in the indexed array
+corresponding to all the true elements in the boolean array. The
+elements in the indexed array is always iterated and returned in
```
Comment (jaimefrio) [inline numpy/doc/indexing.py]: is -> are

**Comment by jaimefrio** [comment]
One minor thing, we should get this merged.

**Comment by rgommers** [comment]
@charris want to fix the typo and merge this?

**Comment by charris** [comment]
OK, done. Thanks for the reminder Ralf.

**Pull request merged**